### DISCUSSION TITLE
Restricted images access from CldImage

### DISCUSSION THREAD

**Discussion by devzaak** [cloudinary-community/next-cloudinary#550] [Q&A]
Hi,
I am struggling to figure out how to access the images set to restricted from my application.
From research it looks like I need to use a server-side `cloudinary` SDK and generate the signed urls when trying to fetch restricted images?
Is there a better way to set this up using only `next-cloudinary` or somewhere I can find best practices for using setting up cloudinary so that original images are unavailable other than through authenticated client.
Only examples I found so far are related to signed uploading of images.

Thank you

**Comment by colbyfayock** [comment]
hey @devzaak thats a good question, there's nothing to support signed URLs out of the box with Next Cloudinary at this time but that's definitely a potential future feature idea.

while you can sign requests without Cloudinary, the Node.js SDK is certainly the easiest and it's actually the recommended path when using the Upload Widget: https://next.cloudinary.dev/clduploadwidget/signed-uploads

that said i would imagine that even generating a signature, i dont think the CldImage component supports taking a signature parameter, so thats a minimum feature that should be added sooner than later
